### ISSUE TITLE
Add metadata description RPC

### ISSUE THREAD

**Issue by apopiak** [paritytech/substrate#8512]
The runtime metadata is currently only returned as a SCALE encoded blob that needs external descriptions to be decoded.
Let's add an RPC endpoint that returns some useful description of the metadata (even if it is created manually short to medium term).

related to #917

**Comment by bkchr** [comment]
What is useful description of the Metadata?

**Comment by shawntabrizi** [comment]
> What is useful description of the Metadata?

Probably a seralized form of JSON in whatever format Polkadot JS needs to connect to the node successfully.

**Comment by bkchr** [comment]
You mean the `types.json`?

**Comment by shawntabrizi** [comment]
Yup. We basically want to bake it into the node, or at least give the option to.

Just one idea.

**Comment by bkchr** [comment]
We could introduce some new metadata version that adds support for adding arbitrary data?

**Comment by xlc** [comment]
> We could introduce some new metadata version that adds support for adding arbitrary data?

https://github.com/paritytech/substrate/issues/3686
